main/src/main/java/SimpleNotePad.java

```diff
@@ -0,0 +1,344 @@
+        uiMenuCopy.addActionListener(this.actionMenuCopy());
+        uiMenuCopy.setActionCommand("copy");
+        uiMenuPaste.addActionListener(this.actionMenuPaste());
+        uiMenuPaste.setActionCommand("paste");
+        uiMenuReplace.addActionListener(this.actionMenuReplace());
+        uiMenuReplace.setActionCommand("replace");
+
+        uiMenuBar.add(uiMenuFile);
+        uiMenuBar.add(uiMenuEdit);
+        setJMenuBar(uiMenuBar);
+        add(new JScrollPane(uiTextPane));
+        setPreferredSize(new Dimension(600,600));
+        setDefaultCloseOperation(JFrame.EXIT_ON_CLOSE);
+        setVisible(true);
+        pack();
+    }
+
+    // Main entry point for the application
+    public static void main(String[] args) {
+        // load config file
+        Properties properties = new Properties();
+        try {
+            File configFile = new File("config.properties");
+            FileReader reader = new FileReader(configFile);
+            properties.load(reader);
+        }
+        catch (Exception e) {
+            // create config file
+            try {
+                properties.setProperty("recent", "");
+                File config = new File("config.properties");
+                FileWriter write = new FileWriter(config);
+                properties.store(write, "settings");
+            }
+            catch (Exception ex) {
+                JOptionPane.showMessageDialog(null,
+                        "Config load error " + ex, "Config load error",
+                        JOptionPane.ERROR_MESSAGE);
+            }
+        }
+
+
+        SimpleNotePad app = new SimpleNotePad(properties);
+    }
+
+    private void UpdateRecents(String newRecent) {
+        try {
+            String propsList = appProperties.getProperty("recent");
+            if (propsList.length() > 0) {
+                String[] propsSplit = propsList.split(",");
+                if (!newRecent.equals(propsSplit[0])) {
+                    if (propsSplit.length > 4) {
+                        String[] newProps = new String[5];
+                        newProps[0] = newRecent;
+                        newProps[1] = propsSplit[0];
+                        newProps[2] = propsSplit[1];
+                        newProps[3] = propsSplit[2];
+                        newProps[4] = propsSplit[3];
+                    } else {
+                        String[] newPropsSplit = new String[propsSplit.length + 1];
+                        for (int i = propsSplit.length; i > 0; i--) {
+                            if (propsSplit[i-1].length() > 0) {
+                                newPropsSplit[i] = propsSplit[i-1];
+                            }
+                        }
+
+                        newPropsSplit[0] = newRecent;
+                        propsSplit = newPropsSplit;
+                    }
+
+                    StringBuilder builder = new StringBuilder();
+
+                    for (String s : propsSplit) {
```

Comment:
What is the purpose of this? Can you comment your code for the logic? It's a little hard to follow.